spec/acceptance/custom_insync_spec.rb

```diff
@@ -0,0 +1,249 @@
+require 'spec_helper'
+require 'tempfile'
+
+RSpec.describe 'a provider using custom insync' do
+  describe 'using `puppet apply`' do
+    subject(:puppet_apply_stdout) do
+      stdout_str, _status = Open3.capture2e("puppet apply --verbose --trace --strict=error --modulepath spec/fixtures -e \"#{manifest}\"")
+      stdout_str
+    end
+
+    context 'when an error is raised during insync?' do
+      let(:manifest) { "test_custom_insync { example: ensure => 'present', case_insensitive_string => 'RaiseError' }" }
+
+      it 'reports the error and change failure' do
+        expect(puppet_apply_stdout).to match %r{Error:}
+        expect(puppet_apply_stdout).to match %r{change from 'FooBar' to 'RaiseError' failed}
+      end
+    end
+
+    context 'when handling subset array comparisons' do
+      let(:manifest) { "test_custom_insync { example: ensure => 'present', some_array => #{test_value} }" }
+
+      context 'when the should array is an exact match for the actual value' do
+        let(:test_value) { %w[a b] }
```

Comment:
Does RSAPI behave the same as "raw" puppet where order is significant (ie it's an array and not a set), if so maybe add a test? Puppet can support set-behavior, but it's up to the type/provider to ignore order, for example the `members` of a Windows `group` are not order dependent: https://github.com/puppetlabs/puppet/blob/3aecb85fd55235d0d5f98170d3268a756f000a17/lib/puppet/provider/group/windows_adsi.rb#L33-L37

Reply:
It does, the missing test is an oversight, I'll get that added - the `test_custom_insync` resource in the fixtures module implements this but tests don't actually validate it.

Reply:
cool thanks!